pkg/amber/provenance.go

```diff
@@ -71,19 +71,59 @@ func validateSLSAProvenanceJSON(provenanceFile []byte) error {
 }
 
 // ParseProvenanceFile reads a JSON file from a given path, and calls ParseStatementData on the
-// content of the file, if the read is successful.
+// content of the file, if the read is successful. It then sets all fields in ProvenanceIR to verify Amber provenances.
 // Returns an error if the file is not a valid provenance statement.
-func ParseProvenanceFile(path string) (*types.ValidatedProvenance, error) {
+func ParseProvenanceFile(path string) (*types.ProvenanceIR, error) {
 	statementBytes, err := os.ReadFile(path)
 	if err != nil {
 		return nil, fmt.Errorf("could not read the provenance file: %v", err)
 	}
-
 	if err := validateSLSAProvenanceJSON(statementBytes); err != nil {
 		return nil, err
 	}
 
-	return types.ParseStatementData(statementBytes)
+	provenanceIR, err := types.ParseStatementData(statementBytes)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse the provenance bytes: %v", err)
+	}
+
+	if err := SetAmberProvenanceData(provenanceIR); err != nil {
+		return nil, fmt.Errorf("could not set the Amber provenance data: %v", err)
+	}
+
+	return provenanceIR, nil
+}
+
+// SetAmberProvenanceData sets data to verify a Amber provenance in the given ProvenanceIR.
+func SetAmberProvenanceData(provenanceIR *types.ProvenanceIR) error {
+	buildType := AmberBuildTypeV1
+
+	predicate, err := slsav02.ParseSLSAv02Predicate(provenanceIR.GetProvenance().Predicate)
```

Comment:
If you make the other changes I suggested, then here you could call `slsav02.ParseProvenanceData` and reuse more of the common functionality between `slsav02` and `amber`.